DuckDuckGo/Bookmarks/Model/BookmarkList.swift

```diff
@@ -44,7 +44,14 @@ struct BookmarkList {
 
     private(set) var allBookmarkURLsOrdered: [IdentifiableBookmark]
     private var favoriteBookmarksOrdered: [IdentifiableBookmark]
-    private var itemsDict: [String: [Bookmark]]
+    private var itemsDict: [String: [Bookmark]] {
+        didSet {
+            updateLowercasedItemsDict()
```

Comment:
Updating lowercased dict here is expensive if `itemsDict` is updated for multiple objects, e.g. in the initializer. Let's call the function as needed in places where `itemsDict` is modified, rather than from here.

Reply:
@ayoy
- `didSet` on properties is not called during `init` methods
- even if it would be called, the `init` is implemented in a way there is a local variable `itemsDict` assigned to the property at the end of the `init`
- other methods have one modifications to the `itemsDict` except updateBookmarkList, which has 2. This should be acceptable.

Reply:
Those are all fair points 👍 thanks!